### PR TITLE
Add versioneer and geos_version

### PR DESCRIPTION
**Pull request by caspervdw**
Following #42  . This doesn't include the release notes.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[versioneer]
+VCS = git
+style = pep440
+versionfile_source = pygeos/_version.py
+versionfile_build = pygeos/_version.py
+tag_prefix =
```
Comment (jorisvandenbossche) [inline setup.cfg]: Compared to what we have in pandas/geopandas, there we also have a `parentdir_prefix` entry. 
But, I don't really understand what this is supposed to do (and tested this branch, which seems to work correctly to set the version and to create sdists)

Comment (caspervdw) [reply setup.cfg]: I tested with and without and saw no difference either.. also not sure what it is about.

**Comment by caspervdw** [comment]
0.4 is out! I took the liberty of copying your release checklist

**Comment by jorisvandenbossche** [comment]
Thanks! And installing it from conda forge is working nicely :)

```(base) joris@joris-XPS-13-9380:~/scipy$ conda create -n test-pygeos python=3.7 pygeos
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/joris/miniconda3/envs/test-pygeos

  added / updated specs:
    - pygeos
    - python=3.7


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    geos-3.7.2                 |       he1b5a44_2         877 KB  conda-forge
    pygeos-0.4                 |   py37hec07ddf_0          42 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         919 KB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main
  bzip2              conda-forge/linux-64::bzip2-1.0.8-h516909a_1
  ca-certificates    conda-forge/linux-64::ca-certificates-2019.9.11-hecc5488_0
  certifi            conda-forge/linux-64::certifi-2019.9.11-py37_0
  geos               conda-forge/linux-64::geos-3.7.2-he1b5a44_2
  libblas            conda-forge/linux-64::libblas-3.8.0-12_openblas
  libcblas           conda-forge/linux-64::libcblas-3.8.0-12_openblas
  libffi             conda-forge/linux-64::libffi-3.2.1-he1b5a44_1006
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0
  libgfortran-ng     pkgs/main/linux-64::libgfortran-ng-7.3.0-hdf63c60_0
  liblapack          conda-forge/linux-64::liblapack-3.8.0-12_openblas
  libopenblas        conda-forge/linux-64::libopenblas-0.3.7-h6e990d7_1
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0
  ncurses            conda-forge/linux-64::ncurses-6.1-hf484d3e_1002
  numpy              conda-forge/linux-64::numpy-1.17.2-py37h95a1406_0
  openssl            conda-forge/linux-64::openssl-1.1.1c-h516909a_0
  pip                conda-forge/linux-64::pip-19.2.3-py37_0
  pygeos             conda-forge/linux-64::pygeos-0.4-py37hec07ddf_0
  python             conda-forge/linux-64::python-3.7.3-h33d41f4_1
  readline           conda-forge/linux-64::readline-8.0-hf8c457e_0
  setuptools         conda-forge/linux-64::setuptools-41.2.0-py37_0
  sqlite             conda-forge/linux-64::sqlite-3.29.0-hcee41ef_1
  tk                 conda-forge/linux-64::tk-8.6.9-hed695b0_1003
  wheel              conda-forge/linux-64::wheel-0.33.6-py37_0
  xz                 conda-forge/linux-64::xz-5.2.4-h14c3975_1001
  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1006


Proceed ([y]/n)? 
...
(base) joris@joris-XPS-13-9380:~/scipy$ conda activate test-pygeos
(test-pygeos) joris@joris-XPS-13-9380:~/scipy$ python
Python 3.7.3 | packaged by conda-forge | (default, Jul  1 2019, 21:52:21) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pygeos import box, area, intersection
>>> polygons_x = box(range(5), 0, range(10, 15), 10)
>>> polygons_y = box(0, range(5), 10, range(10, 15))
>>> area(intersection(polygons_x[:, None], polygons_y[None, :]))
array([[100.,  90.,  80.,  70.,  60.],
       [ 90.,  81.,  72.,  63.,  54.],
       [ 80.,  72.,  64.,  56.,  48.],
       [ 70.,  63.,  56.,  49.,  42.],
       [ 60.,  54.,  48.,  42.,  36.]])
```

**Pull request merged**